### ISSUE TITLE
Confusing error when providing the wrong type to be decoded

### ISSUE THREAD

**Issue by chevdor** [paritytech/jsonrpsee#298]
The following fails:
```
use jsonrpsee::http_client::{traits::Client, HttpClientBuilder, JsonValue, Error};
use std::env;
use tokio::runtime::Runtime;

fn get_http_node() -> String {
	env::var("POLKADOT_HTTP").unwrap_or("http://localhost:9933".to_string())
}

fn main() {
	let url = String::from(get_http_node());
	println!("Connecting to {:?}", &url);
	let code = JsonValue::from("0x3a636f6465"); // :code in hex
	let params = vec![code.into()];

	let rt = Runtime::new().unwrap();
	let client = HttpClientBuilder::default().build(url).expect("Building client");
	let response: Result<Vec<u8>, _> = rt.block_on(client.request("state_getStorage", params.into()));
	println!("response: {:?}", response);

	assert!(response.unwrap().len() > 0);
}
```

with:
```
response: Err(ParseError(Error("missing field `error`", line: 1, column: 3752798)))
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ParseError(Error("missing field `error`", line: 1, column: 3752798))', examples/http-polkadot.rs:51:22
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Debugging the request, I see that we get a big answer from the node.
So I assume the request is sent properly and we get the answer. Decoding however fails at https://github.com/paritytech/jsonrpsee/blob/master/http-client/src/client.rs#L76

**Comment by chevdor** [comment]
@niklasad1 thanks for the hints.

Using the following does not solve the issue but does the trick:
```
let response: Result<String, _> = rt.block_on(client.request("state_getStorage", params.into()));
```

**Comment by jsdw** [comment]
This will hopefully be closed by https://github.com/paritytech/jsonrpsee/pull/853

**Comment by niklasad1** [comment]
Closed by #853
